pandas/core/ops/__init__.py

```diff
@@ -679,6 +668,25 @@ def _frame_arith_method_with_reindex(
     return result.reindex(join_columns, axis=1)
 
 
+def _align_series_as_frame(frame: "DataFrame", series: "Series", axis: int):
+    """
+    If the Series operand is not EA-dtype, we can broadcast to 2D and operate
+    blockwise.
+    """
+    rvalues = series._values
+    if not isinstance(rvalues, np.ndarray):
+        # TODO(EA2D): no need to special-case with 2D EAs
+        return series
+
+    if axis == 0:
+        rvalues = rvalues.reshape(-1, 1)
+    else:
+        rvalues = rvalues.reshape(1, -1)
+
+    rvalues = np.broadcast_to(rvalues, frame.shape)
+    return type(frame)(rvalues, index=frame.index, columns=frame.columns)
```

Comment:
Why do we need to turn this into a DataFrame?

Reply:
The alternative is to re-implement the ensure-shape-shape-values code in ops.blockwise (see the first commit, which did it that way)

Reply:
Why does it need to be the same shape? The block op or array op should be able to handle this broadcasting automatically themselves?

Reply:
ndarray handles broadcasting, but DTA/TDA dont.  I checked with seberg a few months ago who said there isnt a perf penalty to operating on the broadcasted ndarray.

ive got plans to make it so we dont need to wrap the ndarray in a DataFrame (which we actually do in _align_method_FRAME too)

Reply:
>  but DTA/TDA dont.

You mean the "hidden" 2D version of  DTA/TDA? (EAs in general handle broadcasting)

Reply:
> EAs in general handle broadcasting

im not sure thats accurate in general (in particular the `op(ea_len_1, arr_len_n)`), but that can be discussed separately

Reply:
> (in particular the op(ea_len_1, arr_len_n)

Ah, yes, indeed, that's a broadcasting that will not be generally supported (we probably should though, worth to open an issue about that).

Now, I was thinking about the `op(frame_with_EA_column, array_len_n)` case, where the array is aligned on the columns of the frame. Which means that each EA (1 column) gets a scalar, I think (`op(EA, scalar)` ? 
Or, we might want to change this to an array of len 1 in case the scalar loses information? (which can be the case where the 1D array-like is an EA / Series[EA] instead of a numpy array)